pkg/blockdevice/blockdevice.go

```diff
@@ -112,15 +112,17 @@ func (bd *BlockDevice) RereadPartitionTable() error {
 	// Reread the partition table.
 	var err error
 	for i := 0; i < 50; i++ {
-		if _, _, ret := unix.Syscall(unix.SYS_IOCTL, bd.f.Fd(), unix.BLKRRPART, 0); ret != 0 {
-			err = errors.Errorf("re-read partition table: %v", ret)
-			switch ret {
-			case syscall.EBUSY:
-				time.Sleep(100 * time.Millisecond)
-				continue
-			default:
-				return err
-			}
+		if _, _, ret := unix.Syscall(unix.SYS_IOCTL, bd.f.Fd(), unix.BLKRRPART, 0); ret == 0 {
```

Comment:
`ret` will be local to this function in this case. Take a look at https://github.com/talos-systems/talos/blob/a116145c1b1e7763d1d3b866de9e7dd3e5bae990/internal/pkg/mount/mount.go#L100-L110. Might be good to align with that one. The part that makes it work is the early `return nil`.